src/validation.ts

```diff
@@ -68,10 +68,24 @@ export class Validator {
     ) {
     }
 
+    /**
+     * Issue #117: Joi's `validate()` method dies when we pass it our own validation options, so we need to strip it
+     *  out.
+     * @url https://github.com/joiful-ts/joiful/issues/117
+     */
+    protected rectifyOptions(options: ValidationOptions | undefined): [typeof Joi, Joi.ValidationOptions?] {
```

Comment:
nit: This is a little thing but unless there's a specific reason to return a tuple, I would by default return an object with properties. I find returning an object slightly more easier to understand because the properties have names to convey meaning. I do think tuples make sense in some situations, like for example in react hooks you often want to rename the elements returned by `useState`, so returning a tuple keeps it terse. But I don't think it offers much benefit here imho.

Reply:
Fair enough. I thought arrays might be faster, as a micro-optimization, but a quick benchmark test shows they're slower than objects! That'll learn me. 😄

![2020-03-21 13_11_57-Window](https://user-images.githubusercontent.com/6336048/77217348-e412bb80-6b75-11ea-98c1-2d308a0cd94a.png)